src/renderer/dx/CustomTextLayout.h

```diff
@@ -24,10 +24,13 @@ namespace Microsoft::Console::Render
                          gsl::not_null<IDWriteTextAnalyzer1*> const analyzer,
                          gsl::not_null<IDWriteTextFormat*> const format,
                          gsl::not_null<IDWriteFontFace1*> const font,
-                         const std::basic_string_view<::Microsoft::Console::Render::Cluster> clusters,
                          size_t const width,
                          IBoxDrawingEffect* const boxEffect);
 
+        [[nodiscard]] HRESULT STDMETHODCALLTYPE AppendClusters(const std::basic_string_view<::Microsoft::Console::Render::Cluster> clusters);
+
+        [[nodiscard]] HRESULT STDMETHODCALLTYPE Reset();

```

Comment:
A `Reset` is being done, both times `AppendClusters` is called. Would it make sense to have a `SetClusters` instead which does both?

Reply:
My mental model has this row:

60 chars red, 20 chars green, 20 chars red, 10 chars blue, 10 chars red.

Instead of calling Draw 5 times, I'd love to call it 3 times, one per color.

My theory is that I will make AppendClusters take the color and then batch together everything of the same color to reduce DrawGlyphRun calls and save the command queue further.

Reply:
(But if I don't get there, then yes, your suggestion makes way more sense.)

Reply:
Okay, this idea will be carried forward in a 2nd PR, going to take this one as is and leave this as is so I can change it easily.